driver/mloLayerNormHost.hpp

```diff
@@ -0,0 +1,194 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifdef MIOPEN_BETA_API
+#ifndef MLO_LAYERNORMHOST_H_
+#define MLO_LAYERNORMHOST_H_
+
+////////////////////////////////////////////////////////////
+//
+///////////////////////////////////////////////////////////
+
+template <typename Tgpu, typename Tcheck /* the data type used in CPU checkings (usually double) */>
+int mloLayerNormForwardRunHost(miopenTensorDescriptor_t inputDesc,
+                               miopenTensorDescriptor_t weightDesc,
+                               miopenTensorDescriptor_t biasDesc,
+                               miopenTensorDescriptor_t outputDesc,
+                               miopenTensorDescriptor_t meanDesc,
+                               miopenTensorDescriptor_t rstdDesc,
+                               Tgpu* input,
+                               Tgpu* weight,
+                               Tgpu* bias,
+                               Tcheck* outputhost,
+                               Tcheck* meanhost,
+                               Tcheck* rstdhost,
+                               double eps,
+                               int normalized_dim,
+                               miopenLayerNormMode_t mode)
+{
+    auto dims         = miopen::deref(inputDesc).GetLengths();
+    size_t grid_size  = 1;
+    size_t outer_size = 1;
+    size_t inner_size = 1;
+    size_t i          = 0;
+    for(; i < normalized_dim; i++)
+    {
+        outer_size *= dims[i];
+        grid_size *= dims[i];
+    }
+
+    for(; i < dims.size(); i++)
+    {
+        inner_size *= dims[i];
+        grid_size *= dims[i];
+    }
+
+    int ret = 0;
+
+    for(int o = 0; o < outer_size; o++)
+    {
+        double pmean = 0;
+        double pvar  = 0;
+        for(i = 0; i < inner_size; i++)
+        {
+            double tmp = input[o * inner_size + i];
+            pmean += tmp;
+            pmean += tmp * tmp;
```

Comment:
Probably you've meant
```suggestion
            pvar += tmp * tmp;
```
?

Reply:
I have corrected that part. 
There was a problem in calling the HIP kernel, so it was uploaded as a draft. 
Now that the problem is solved, I'm going to try to solve the review one by one.
thank you.